### PR TITLE
Refactor calloc

### PR DESCRIPTION
**Pull request by fortizc**
Refactor all `calloc` calls used to assign memory to `char` pointers.

### CODE REVIEW

File: metadata.c [hunk 1]
```diff
@@ -403,15 +403,15 @@ static int pv_usermeta_parse(struct pantavisor *pv, char *buf)
 		n = (*key_i)->end - (*key_i)->start;
 
 		// copy key
-		key = calloc(1, (n+1) + sizeof(char*));
+    key = calloc(n + 1 + 1, sizeof(char));
```
Comment (anibalportero) [inline metadata.c:406]: is this 1+1 needed=?

File: metadata.c [hunk 2]
```diff
@@ -403,15 +403,15 @@ static int pv_usermeta_parse(struct pantavisor *pv, char *buf)
...
 		if (!key)
 			break;
 
 		strncpy(key, um+(*key_i)->start, n);
 
 		// copy value
 		n = (*key_i+1)->end - (*key_i+1)->start;
-		value = calloc(1, (n+1) + sizeof(char*));
+		value = calloc(n + 1 + 1, sizeof(char));
```
Comment (anibalportero) [inline metadata.c:414]: same here

File: pantahub.c [hunk 3]
```diff
@@ -152,7 +152,7 @@ const char** pv_ph_get_certs(struct pantavisor *__unused)
 		return NULL;
 
 	// Always n-1 due to . and .., and need one extra
-	cafiles = calloc(1, (sizeof(char*) * (n-1)));
+	cafiles = calloc(n - 1, sizeof(char*));
```
Comment (anibalportero) [inline pantahub.c:155]: I guess just char here to keep consistence

File: metadata.c (hunk 1 above)
Comment (asac) [reply metadata.c:406]: also: indent

File: pantahub.c (hunk 3 above)
Comment (asac) [reply pantahub.c:155]: cafiles is an array of char* no? looks correct to me ...

File: state.c [hunk 4]
```diff
@@ -89,7 +89,7 @@ struct pv_state* pv_state_new(const char *rev, state_spec_t spec)
 
 	s = calloc(1, sizeof(struct pv_state));
 	if (s) {
-		s->rev = calloc(1, len * sizeof(char*));
+    s->rev = calloc(len, sizeof(char));
```
Comment (asac) [inline state.c:92]: indent

File: pantahub.c (hunk 3 above)
Comment (fortizc) [reply pantahub.c:155]: yes, cafiles is a array of char*

Comment (anibalportero) [reply pantahub.c:155]: oh, sorry. nevermind

File: metadata.c (hunk 1 above)
Comment (fortizc) [reply metadata.c:406]: @anibalportero my bad, yes looking at the code `+ 1` is enough my bad
@asac I will fix it

Comment (fortizc) [reply metadata.c]: Fixed in 95ec83b7d8abe1e192ecc840cdfc54a6069f59dd

**Pull request merged**